packages/envied_generator/lib/src/generate_field.dart

```diff
@@ -10,11 +10,51 @@ import 'package:source_gen/source_gen.dart';
 /// Since this function also does the type casting,
 /// an [InvalidGenerationSourceError] will also be thrown if
 /// the type can't be casted, or is not supported.
-Iterable<Field> generateFields(FieldElement field, String value) {
+Iterable<Field> generateFields(
+  FieldElement field,
+  String? value, {
+  required bool allowOptional,
+}) {
   final String type = field.type.getDisplayString(withNullability: false);
 
   late final Expression result;
 
+  if (allowOptional && value == null) {
```

Comment:
You might want to merge both this `if` condition and the one below it into one block, i.e.

```dart
if (value == null) {
  if (allowOptional) {
    /// your new stuff
  }

  throw InvalidGenerationSourceError(
    'Environment variable not found for field `${field.name}`.',
    element: field,
  );
}